### PR TITLE
Worldgen move3

### PR DESCRIPTION
**Pull request by DimpyRed**
I still have some cleanup to do, but this is the basic structure.

### CODE REVIEW

File: common/Cargo.toml [hunk 1]
```diff
@@ -20,6 +19,8 @@ fxhash = "0.2.1"
 tracing = "0.1.10"
 hecs = "0.2.9"
 tracing-subscriber = { version = "0.2.5", default-features = false, features = ["env-filter", "smallvec", "fmt", "ansi", "chrono", "parking_lot"] }
+rand = "0.7.3"
+rand_pcg = "0.2.1"
```
Comment (Ralith) [inline common/Cargo.toml:23]: Can we also remove this dependency from `client`?

File: common/src/chunk.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+use crate::lru_slab::SlotId;
+use crate::node::VoxelData;
+
+pub enum Chunk {
```
Comment (Ralith) [inline common/src/chunk.rs:4]: I'd dump this into the `node` module, since it's so tiny.

File: common/src/node.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    pub fn get(&self, index: usize) -> Material {
+        match *self {
+            VoxelData::Dense(ref d) => d[index],
+            VoxelData::Solid(mat) => mat,
+        }
+    }
+}
+
+
+use crate::lru_slab::SlotId;
```
Comment (Ralith) [inline common/src/node.rs:43]: This should be at the top with the other `use`s

File: common/src/node.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+pub struct Node {
+    pub state: NodeState,
+    /// We can only populate chunks which lie within a cube of populated nodes, so nodes on the edge
+    /// of the graph always have some `None` chunks.
+    pub chunks: Chunks<Chunk>,
+}
+
+#[derive(PartialEq)]
+pub enum VoxelData {
```
Comment (Ralith) [inline common/src/node.rs:18]: Let's sort this type (and its `impl` block) below `Chunk`, since that's where it's used.

Comment (DimpyRed) [reply common/src/node.rs:34]: This is what you were asking for, right?

Comment (Ralith) [reply common/src/node.rs:34]: Yep, thanks!

**Comment by Ralith** [comment]
Thanks! This will make a great foundation for server-side physics and should make testing and iterating on the worldgen code more convenient.

**Pull request merged**